### PR TITLE
All set of changes to make Api_t/Simple_t/Daemon_t unit test to work

### PR DESCRIPTION
**Pull request by vkuznet**
PLEASE DO NOTE MERGE until the work is done

### CODE REVIEW

**Comment by ericvaandering** [comment]
retest this please

**Comment by ericvaandering** [comment]
Ok, @vkuznet this should be your limited test coming up. There are a couple of things from Alan ahead of you in the queue, so it may be a while before it completes.

File: test/python/WMCore_t/REST_t/Api_t.py [hunk 1]
```diff
@@ -1,18 +1,27 @@
-from cherrypy.test import helper
-from cherrypy import expose, response, config as cpconfig
+# system modules
+import cjson, re, zlib
```
Comment (ericvaandering) [inline test/python/WMCore_t/REST_t/Api_t.py]: I think json is every bit as good as cjson now (or actually is cjson)

Comment (vkuznet) [inline test/python/WMCore_t/REST_t/Api_t.py]: I can change it to json, but I think it needs to be clarified by Seangchan.

**Comment by vkuznet** [comment]
Seangchan, as far as I concern the changes are in good shape, but I didn't touch cjson since it was present in a code. Therefore you can merge these changes or let me know if you want me to convert cjson to json and then we'll merge this PR.

**Comment by ericvaandering** [comment]
Let's go as is. I missed that cjson was original.

Sent from a mobile device. Please excuse my brevity or transcription errors.

On Sep 8, 2015, at 10:59, Valentin Kuznetsov <notifications@github.com<mailto:notifications@github.com>> wrote:


Seangchan, as far as I concern the changes are in good shape, but I didn't touch cjson since it was present in a code. Therefore you can merge these changes or let me know if you want me to convert cjson to json and then we'll merge this PR.

**Comment by ticoann** [comment]
Is json is as good as cjson in python 2.6.8? Or after python 2.7?
@vkuznet, this can be merged right? (Just double checking :smile: )

**Comment by vkuznet** [comment]
The cjson is still better performance wise, and, yes we can merge this one.

**Comment by ericvaandering** [comment]
I might have been confusing simplejson with json (lots of confusion to be had) but this suggests there is little difference anymore: http://stackoverflow.com/questions/706101/python-json-decoding-performance

Takeaway is we shouldn't be writing any new code that uses anything other than the standard library.

**Comment by vkuznet** [comment]
Eric,
you may read this thread on python mailing list I participated long time ago
http://bugs.python.org/issue6594
The speed is not an issue among the two and rather the memory footprint,
especially loading nested JSON's such as provided by PhEDEx APIs.

But I agree apart from some corner cases we should/must stick to standard
library.
Valentin.

**Comment by ticoann** [comment]
One other thing I remember was cjson has bug which convert all the unit code to str. However that actually served as a feature in WMCore. So it might break some of the code in WMCore if we switch to json.  Yes I agree to move to standard library as well unless there is big advantage using cjson

**Comment by PerilousApricot** [comment]
Not only that, JSON requires that all strings (including dict keys) are unicode. cjson (used to, at least) convert dict keys to 'string' and keep all other strings as unicode, which was a catastrophe when we put it against the HTTP frontend validation.

**Comment by ericvaandering** [comment]
Fortunately our usage seems to be about 80% json, 10% cjson, and 10% simplejson.

**Comment by vkuznet** [comment]
Simplejson should just go, out of questions.

**Comment by PerilousApricot** [comment]
Oh yeah. All of the simplejson/WMCore.JsonWrapper stuff is almost certainly legacy from when we were on 2.4 and json wasn't in stdlib.

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Wrappers/JsonWrapper/__init__.py

**Comment by ericvaandering** [comment]
Yeah, I know. We might even have an issue to get rid of that crap. If not, we should

**Comment by ericvaandering** [comment]
#5785

**Pull request merged**